synapse/federation/sender/transaction_manager.py

```diff
@@ -44,93 +52,107 @@ def __init__(self, hs):
     @defer.inlineCallbacks
     def send_new_transaction(self, destination, pending_pdus, pending_edus):
 
-        # Sort based on the order field
-        pending_pdus.sort(key=lambda t: t[1])
-        pdus = [x[0] for x in pending_pdus]
-        edus = pending_edus
+        # Make a transaction-sending opentracing span. This span follows on from
+        # all the edus in that transaction. This needs to be done since there is
+        # no active span here, so if the edus were not received by the remote the
+        # span would have no causality and it would be forgotten.
 
-        success = True
+        span_contexts = (
```

Comment:
tbh it would probably good to note in the comment above that this is (deliberately) a generator so won't get evaluated unless opentracing is enabled